src/components/input/input.tsx

```diff
@@ -692,7 +692,7 @@ export class Input implements LabelableComponent, FormComponent {
         ? localizeNumberString(newValue, this.locale, this.groupSeparator)
         : "";
 
-    this.internalValueChange = origin === "internal";
+    this.internalValueChange = origin === "internal" && this.value !== newValue;
```

Comment:
It could, yes.  I think I wrote this as a one-liner while I was helping Eliza out, but sure it could be a check at the top of the function that just returns when the value isn't changing.